pkg/util/kubernetes/kubelet/containers.go

```diff
@@ -113,6 +115,39 @@ func parseContainerInPod(status ContainerStatus, pod *Pod) (*containers.Containe
 	return c, nil
 }
 
+func parseContainerNetworkAdresses(status ContainerStatus, pod *Pod) []containers.NetworkAddress {
+	addrList := []containers.NetworkAddress{}
+	podIP := net.ParseIP(pod.Status.PodIP)
+	if podIP == nil {
+		log.Warnf("Unable to parse pod IP: %v for pod: %s", pod.Status.PodIP, pod.Metadata.Name)
+		return addrList
+	}
+	hostIP := net.ParseIP(pod.Status.HostIP)
+	if hostIP == nil {
+		log.Warnf("Unable to parse host IP: %v for pod: %s", pod.Status.HostIP, pod.Metadata.Name)
+		return addrList
+	}
+	// Look for the ports in container spec
+	for _, s := range pod.Spec.Containers {
+		if s.Name == status.Name {
+			for _, port := range s.Ports {
+				address := containers.NetworkAddress{Protocol: port.Protocol}
+				if port.HostPort > 0 {
+					address.IP = hostIP
+					address.Port = port.HostPort
+				} else if port.ContainerPort > 0 {
```

Comment:
IIRC a port can still be accessible on the container IP when it is bound to a host port, should both be added to the addrList?

Reply:
Good point, let's add both 👍